src/main/java/org/opentripplanner/apis/gtfs/mapping/RouteRequestMapper.java

```diff
@@ -283,132 +110,616 @@ private static Stream<String> parseOperation(Collection<Map<String, Collection<S
+  ) {
+    if (args != null) {
+      var unpreferredCost = args.getGraphQLUnpreferredCost();
+      if (unpreferredCost != null) {
+        preferences.withUnpreferredVehicleParkingTagCost(unpreferredCost.toSeconds());
+      }
+      var filters = getParkingFilters(environment, "car");
+      preferences.withRequiredVehicleParkingTags(parseSelectFilters(filters));
+      preferences.withBannedVehicleParkingTags(parseNotFilters(filters));
+      var preferred = getParkingPreferred(environment, "car");
+      preferences.withPreferredVehicleParkingTags(parseSelectFilters(preferred));
+      preferences.withNotPreferredVehicleParkingTags(parseNotFilters(preferred));
+    }
+  }
 
-    private final DataFetchingEnvironment environment;
+  private static void setCarRentalPreferences(
+    VehicleRentalPreferences.Builder preferences,
+    GraphQLTypes.GraphQLCarRentalPreferencesInput args
+  ) {
+    if (args != null) {
+      var allowedNetworks = args.getGraphQLAllowedNetworks();
+      if (allowedNetworks != null && allowedNetworks.size() > 0) {
+        preferences.withBannedNetworks(Set.copyOf(allowedNetworks));
+      }
+      var bannedNetworks = args.getGraphQLBannedNetworks();
+      if (bannedNetworks != null && bannedNetworks.size() > 0) {
+        preferences.withBannedNetworks(Set.copyOf(bannedNetworks));
+      }
+    }
+  }
 
-    public CallerWithEnvironment(DataFetchingEnvironment e) {
-      this.environment = e;
+  private static void setScooterPreferences(
+    ScooterPreferences.Builder preferences,
+    GraphQLTypes.GraphQLScooterPreferencesInput args
+  ) {
+    if (args != null) {
+      var speed = args.getGraphQLSpeed();
+      if (speed != null) {
+        preferences.withSpeed(speed);
+      }
+      var reluctance = args.getGraphQLReluctance();
+      if (reluctance != null) {
+        preferences.withReluctance(reluctance);
+      }
+      preferences.withRental(rental -> setScooterRentalPreferences(rental, args.getGraphQLRental())
+      );
+      setScooterOptimization(preferences, args.getGraphQLOptimization());
     }
+  }
 
-    private static <T> void call(
-      DataFetchingEnvironment environment,
-      String name,
-      Consumer<T> consumer
-    ) {
-      if (!name.contains(".")) {
-        if (hasArgument(environment, name)) {
-          consumer.accept(environment.getArgument(name));
-        }
-      } else {
-        String[] parts = name.split("\\.");
-        if (hasArgument(environment, parts[0])) {
-          Map<String, T> nm = environment.getArgument(parts[0]);
-          call(nm, String.join(".", Arrays.copyOfRange(parts, 1, parts.length)), consumer);
+  private static void setScooterRentalPreferences(
+    VehicleRentalPreferences.Builder preferences,
+    GraphQLTypes.GraphQLScooterRentalPreferencesInput args
+  ) {
+    if (args != null) {
+      var allowedNetworks = args.getGraphQLAllowedNetworks();
+      if (allowedNetworks != null && allowedNetworks.size() > 0) {
```

Comment:
This check happens over and over. Can you extract something like `CollectionUtils.hasValue`?

Reply:
Empty array is returned (which makes sense). I've used the utils now.